deps/spidershim/test/value.cc

```diff
@@ -32,8 +32,22 @@ static int StrNCmp16(uint16_t* a, uint16_t* b, int n) {
   }
 }
 
+// Based on the function of the same name in test-api.cc, but modified
+// to compare strings in a SpiderShim-compatible way.
+static bool SameSymbol(Local<String> s1, Local<String> s2) {
+  Isolate* isolate = Isolate::GetCurrent();
+  // Use Persistent<T> to get access to the protected Value "equal to" operator.
```

Comment:
> Hmm, I'm not sure if I understand the assumption behind this code. Can you please explain what this is trying to ensure? It's probably worth clarifying the comment here...

It ensures that the JSString object referenced by the first Local<String> is the same JSString object referenced by the second Local<String>, i.e. that both Local<String>s reference the same JSString (as they should do when the strings were interned) rather then each referencing a different JSString (as they should do when the strings weren't interned).

I've expanded the commentary (and added a test to confirm that SameSymbol returns false for non-interned strings) in 99faddb.